src/test/java/com/faforever/client/config/FafClientApplicationTest.java

```diff
@@ -30,6 +31,7 @@ public void testDoesItSmoke() throws Exception {
         context.getBeanFactory().registerSingleton("hostService", mock(PlatformService.class));
         context.getBeanFactory().registerSingleton("stage", new Stage());
         context.getBeanFactory().registerSingleton("uidService", mock(UidService.class));
+        context.register(ForgedAlliancePrefs.class);
```

Comment:
noooooooo! Do not register the Preference classes they need to be obtained by calling preferenceService.getpreferences.ForgedAlliancePrefs();

Spring creates a new instance if u register it. an empty one but the those are the settings loaded fom a json file. 

Any further if u would need to register it u would need to annotate it with @Component (like the ForgedAlliancePrefs.class) 

ForgedAlliancePrefs can not be Injected by Spring or better it hould not be

Reply:
I see and I understand it. I fixed that for FafClientApplication but missed to align the test accordingly.